### ISSUE TITLE
[FEAT] Calculate term frequency (TF) weights using only one specified data set, or using a pre-specified TF table

### ISSUE THREAD

**Issue by samkodes** [moj-analytical-services/splink#1992]
### Is your proposal related to a problem?

I am working on a linking problem in which individuals in dataset 1 appear multiple times (with variant names, addresses, etc - records themselves are not duplicated) but individuals in dataset 2 have a unique record.  I would like to find the single best match for each record from dataset 1 in dataset 2 (in theory, every individual in dataset 1 should have a unique master record in dataset 2).

In this context the current method of calculating term frequencies, which I believe simply stacks the two data sets together, is dominated by the frequencies in dataset 1 (the bigger dataset). However, the multiplicity of individuals in dataset 1 varies widely, and is irrelevant to the probability of a match, conditional on any particular row in dataset 1. 

More broadly, I believe that in general a "dedupe" approach that uses TF adjustments calculated from duplicative data may be unreliable to the extent that some records are duplicated more than others so the term frequencies calculated may not be representative of underlying population term frequencies. 

### Describe the solution you'd like

I believe in my scenario TF adjustments can be carried out as usual to improve matching accuracy, **provided that term frequencies are calculated using only dataset 2**. I may have to check the math on that, but it's my first guess; approaching the calculation **conditional on a given row in dataset 1** gives the same final adjustment to match weights as described in the guide on [Term Frequency Adjustments](https://moj-analytical-services.github.io/splink/topic_guides/comparisons/term-frequency.html) because m_x =m and the quadratic T_x term in u_x becomes a single T_x term instead.  So the Bayes factor is unchanged from the usual case. If the m_x and u_x terms are used directly in the code, they can each simply drop a factor of T_x.  

In the more general scenario, when the datasets being considered are both duplicative but population term frequencies are known, they could be supplied by the user via a user-defined table. Conditioning on the left element again simplifies the analysis. In this case the conditioning on non-match status in u_x should not be considered in terms of the data distribution, but instead in terms of a hypothetical population, with each comparison (right) element considered to be a draw from that hypothetical population; multiplicities in the data set can be considered sampling quirks. 

In conclusion, I propose allowing users to provide a pre-specified TF table for each TF column, or to specify a single data set to use for calculation of TFs.

**Comment by RobinL** [comment]
You can use https://github.com/moj-analytical-services/splink/blob/dd2df284d0139cfa639a99745981a52994a8281f/splink/linker.py#L3831

To find out the format the table needs to be in, use https://github.com/moj-analytical-services/splink/blob/dd2df284d0139cfa639a99745981a52994a8281f/splink/linker.py#L1237
